df_chat/tests/test_message_endpoint.py

```diff
@@ -26,6 +31,110 @@ def test_message_creation_success(self):
+
+        # * 1. Keyword search - ok cases
+        for body in [
+            "Test message 1234.",
+            "Test message 1234",
+            "Test",
+            "test",
+            "Message",
+            "message",
+            "1234",
+        ]:
+            messages_list_response = await self.async_client.get(
+                messages_list_api_url,
+                {"body": body},
+            )
+            # Assert the correct HTTP status code for response.
+            self.assertEqual(messages_list_response.status_code, http.HTTPStatus.OK)
+            # Assert the response body contains the keyword.
+            self.assertIn(
+                body.lower(), messages_list_response.json()[0]["body"].lower()
+            )
+
+        # * 1. Keyword search - not ok cases
+        for body in [
+            "somethingrandom",
+            "messages",
+            "124",
+        ]:
+            messages_list_response = await self.async_client.get(
+                messages_list_api_url,
+                {"body": body},
+            )
+            # Assert the correct HTTP status code for response.
+            self.assertEqual(messages_list_response.status_code, http.HTTPStatus.OK)
+
+        # * 2. username search - ok case
+        messages_list_response = await self.async_client.get(
+            messages_list_api_url,
+            {"username": user.username},
+        )
+        # Assert the correct HTTP status code for response.
+        self.assertEqual(messages_list_response.status_code, http.HTTPStatus.OK)
+        # Assert the message `room_user_id` matches the `id` field of the `RoomUser` object
+        #   corresponding to the user.
+        room_user_obj = RoomUser.objects.get(user__id=user.id)
+        self.assertEqual(
+            messages_list_response.json()[0]["room_user_id"], room_user_obj.id
+        )
+
+        # * 2. username search - not ok case
+        messages_list_response = await self.async_client.get(
+            messages_list_api_url,
+            {"username": "invalid_username"},
+        )
+        # Assert the correct HTTP status code for response.
+        self.assertEqual(messages_list_response.status_code, http.HTTPStatus.OK)
+
+        now = datetime.now()
+        before = now - timedelta(hours=1)
+        datetime_format = "%Y-%m-%dT%H:%M:%S.%f"
+
+        # * datetime range search - range start
+        messages_list_response = await self.async_client.get(
+            messages_list_api_url,
+            {"created_gte": str(before)},
+        )
+        message_created = datetime.strptime(
+            messages_list_response.json()[0]["created"], datetime_format
+        )
+        # Assert the correct HTTP status code for response.
+        self.assertEqual(messages_list_response.status_code, http.HTTPStatus.OK)
+        # Assert message `created` datetime is larger than `before` datetime.
+        self.assertTrue(message_created >= before)
```

Comment:
While running this test, the messages are always created in the past few minutes/seconds. So, this test will always pass.
Please add a message to the database that appears to have been created over an hour ago and check that the message is not returned in the response.

Reply:
I made sure of this. The `datetime` format of the messages are `%Y-%m-%dT%H:%M:%S.%f` and event though the message creation and the list API call are close to each other, the time difference in the `%f` portion of the `created` field asserts that the message is indeed created before the API call.